### ISSUE TITLE
Add analogy of fail fast to Pipeline definition to cancel in case of 1 failed task

### ISSUE THREAD

**Issue by andrejusc** [tektoncd/pipeline#6313]
### Feature request

Have within Pipeline's definition explicit field to say if whole pipeline should be cancelled in case of 1 task had failed.

### Use case

Having multiple tasks running in parallel, where some could be short and some could be long in execution time, it would be beneficial to have ability to cancel whole pipeline if some short task failed already.

Looking at this documentation https://tekton.dev/docs/pipelines/pipelineruns/#gracefully-cancelling-a-pipelinerun - it seems to possible to achieve from outside by changing PipelineRun status. But here feature request is for PipelineRun to manage such cancel condition itself if specified upfront in Pipeline's definition.

**Comment by chengjoey** [comment]
Set a field in the pipelinerun to indicate that if one task fails, other parallel tasks should be cancelled directly. This seems meaningful because in cases where it is known that the pipelinerun will fail, resources can be released as soon as possible without having to manually cancel the entire pipelinerun from outside.

**Comment by andrejusc** [comment]
/remove-lifecycle stale

I think such requires some effort to be earlier than later in place.

**Comment by vdemeester** [comment]
Closing as duplicate of #1684 (Failure Strategy for TaskRuns in a PipelineRun), which is the comprehensive design discussion for fail-fast and other failure strategies in Pipelines (60 comments).

The "cancel pipeline on first failed task" behavior is part of the broader failure strategy design tracked there.
